### PR TITLE
Issue templating

### PR DESCRIPTION
**Pull request by pranshugupta54**
**What kind of change does this PR introduce?**

**Issue Number:**

Ref #1774

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
Update the PR template too if this is possible using this methodology

**Comment by varshith257** [comment]
@palisadoes Please check the files for any suggestions for improvements of descriptions of placeholders in both files to make it more readable

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  - type: textarea
+    id: actual
+    attributes:
+      label: Actual behavior
+      description: A clear and concise description of how the code performed w.r.t expectations.
+      placeholder: What should have happened?
+    validations:
+      required: true
+  - type: textarea
+    id: screeshot
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/bug_report.yml:48]: typo?

**Review by palisadoes — CHANGES_REQUESTED** [review]

**Comment by palisadoes** [comment]
I'm not familiar with the syntax for this. 
1. Are the checkbox values going to be represented by a `Yes` or `No` value on submission?

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      url: https://github.com/PalisadoesFoundation/talawa-docs/issues/new/choose
+      about: This issue tracker is not for Documentation issues. Please file Documentation issues here.
+    - name: 📃 Mobile Issue
+      url: https://github.com/PalisadoesFoundation/talawa/issues/new/choose
+      about: This issue tracker is not for Mobile issues. Please file Mobile issues here.
+    - name: 📃 API Issue
+      url: https://github.com/PalisadoesFoundation/talawa-api/issues/new/choose
+      about: This issue tracker is not for API issues. Please file API issues here.
+    - name: 🤔 Questions and Help
+      url: https://join.slack.com/t/thepalisadoes-dyb6419/shared_invite/zt-2e75w7ups-AVCObKnnHqNzd76CmIRyag
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/config.yml:12]: The slack link changes every 30 days. Use a link to:
- https://github.com/PalisadoesFoundation/talawa

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+contact_links:
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/config.yml:1]: 1. What is this file used for?
2. Where is documentation on this templating methodology?

**Review by palisadoes — CHANGES_REQUESTED** [review]

**Comment by pranshugupta54** [comment]
> I'm not familiar with the syntax for this.
> 
> 1. Are the checkbox values going to be represented by a `Yes` or `No` value on submission?

@palisadoes, you can test it here https://github.com/pranshugupta54/talawa-docs/issues/new/choose

**Comment by varshith257** [comment]
> I'm not familiar with the syntax for this. 
> 1. Are the checkbox values going to be represented by a `Yes` or `No` value on submission?

@palisadoes Check out his changes here :
https://github.com/pranshugupta54/talawa-docs/issues/new/choose

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (pranshugupta54) [reply .github/ISSUE_TEMPLATE/config.yml:1]: ## 1) It's adding links to other resources
image

## 2) https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

Comment (pranshugupta54) [reply .github/ISSUE_TEMPLATE/config.yml:1]: You can also try to edit the yml file and it automatically gives the complete formatting on right side:
-- 
image

**Comment by varshith257** [comment]
@palisadoes Have a look at it

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (palisadoes) [reply .github/ISSUE_TEMPLATE/config.yml:1]: Remove all the links to creating issues in other repositories. They are not relevant

**Review by palisadoes — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+    - name: 📃 Blank Issue
+      url: https://github.com/PalisadoesFoundation/talawa-admin/issues/new
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/config.yml:3]: No blank issue creation. We need a version of the existing template text that will cover both bugs and features.
The use of the guiding text has been very useful in creating issues and PRs that are easier to understand due to the better documentation

**Review by palisadoes — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/config.yml (hunk 4 above)
Comment (pranshugupta54) [reply .github/ISSUE_TEMPLATE/config.yml:3]: For the pre-populated text, we currently have bug and feature, how do I implement them in one? or do i make 2 blank

Comment (pranshugupta54) [reply .github/ISSUE_TEMPLATE/config.yml]: Added existing and new templates - [check here](https://github.com/pranshugupta54/talawa-admin/issues/new/choose)

image

**Comment by palisadoes** [comment]
1. This is too many options. Old and New
2. We need a single custom option of using pre-populated text like the ones we have now. Text that would cover both features and bugs without having separate feature and bug sections. It'll have to be generic

**Comment by pranshugupta54** [comment]
> 1. This is too many options. Old and New
> 2. We need a single custom option of using pre-populated text like the ones we have now. Text that would cover both features and bugs without having separate feature and bug sections. It'll have to be generic

Implemented a generic template:
--
image

Pre-populated text:
--
image

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
+name: 🐞 Bug Report (NEW)
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/bug_report.yml:1]: Remove (NEW)

File: .github/ISSUE_TEMPLATE/config.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+contact_links:
+    - name: 🤔 Questions and Help
+      url: https://github.com/PalisadoesFoundation/talawa?tab=readme-ov-file#talawa
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/config.yml:3]: Remove. It's not necessary. We need fewer options

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 7]
```diff
@@ -0,0 +1,69 @@
+---
+name: 🛠️ Feature Request (NEW)
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/feature_request.yml:2]: Remove (NEW)

File: .github/ISSUE_TEMPLATE/issue.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+---
+name: Issue Report / Feature Request
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/issue.md:2]: Change to "Custom"

File: .github/ISSUE_TEMPLATE/issue.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+about: Create a report to help us improve or suggest an idea for this project.
+title: Issue Report / Feature Request
+---
+
+**Type**:
+- [ ] Issue Report
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/issue.md:8]: Change to `Bug Report` for consistency

File: .github/ISSUE_TEMPLATE/issue.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+---
+name: Issue Report / Feature Request
+about: Create a report to help us improve or suggest an idea for this project.
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/issue.md:3]: Change to "Create bug reports and feature requests using a customizable form"

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by pranshugupta54** [comment]
Fixed all

**Comment by palisadoes** [comment]
I'm going to close this. We have more pressing priorities.

**Pull request closed without merging**